cognite/client/data_classes/data_modeling/ids.py

```diff
@@ -1,15 +1,24 @@
 from __future__ import annotations
 
+from abc import ABC
 from dataclasses import asdict, dataclass, field
-from typing import ClassVar, Literal, Optional, Sequence, Tuple, Type, TypeVar, Union, cast
+from typing import Any, ClassVar, Literal, Optional, Sequence, Tuple, Type, TypeVar, Union, cast
 
 from cognite.client.utils._auxiliary import rename_and_exclude_keys
 from cognite.client.utils._identifier import DataModelingIdentifier, DataModelingIdentifierSequence
 from cognite.client.utils._text import convert_all_keys_recursive, convert_all_keys_to_snake_case
 
 
 @dataclass(frozen=True)
-class DataModelingId:
+class AbstractDataclass(ABC):
```

Comment:
What is the advantage of a shared base class?

Reply:
Ensure frozen?

Reply:
Ensuring that the direct subclasses of this class aren't instantiated directly. For Python to validate this at runtime without this, you need at least one abstract method defined (which we don't have here).